packages/chevrotain/src/parse/parser/traits/tree_builder.ts

```diff
@@ -41,6 +62,62 @@ export class TreeBuilder {
             this.manySepFirstInternal = this.manySepFirstInternalNoCst
             this.atLeastOneSepFirstInternal = this.atLeastOneSepFirstInternalNoCst
         }
+
+        // TODO: case insensitive options are an awesome idea.
+        //       but we should be consistent so if we do this we should apply it at the lexer level too
+        //       or not at all.
+        if (/full/i.test(this.nodePositionTracking)) {
+            this.setNodeLocation = setNodeLocationFull
+            this.setInitialNodeLocation = this.setInitialNodeLocationFull
+        } else if (/onlyStart/i.test(this.nodePositionTracking)) {
+            this.setNodeLocation = setNodeLocationOnlyStart
+            this.setInitialNodeLocation = this.setInitialNodeLocationOnlyStart
+        } else if (/onlyOffset/i.test(this.nodePositionTracking)) {
+            this.setNodeLocation = setNodeLocationOnlyOffset
+            this.setInitialNodeLocation = this.setInitialNodeLocationOnlyOffset
+        } else if (/none/i.test(this.nodePositionTracking)) {
+            this.setNodeLocation = NOOP
+            this.setInitialNodeLocation = NOOP
+        } else {
+            throw Error(
+                `Invalid <nodePositionTracking> config option: "${
+                    config.nodePositionTracking
+                }"`
+            )
+        }
+    }
+
+    setInitialNodeLocationOnlyOffset(cstNode: CstNode): void {
+        cstNode.location = {
+            startOffset: Infinity
```

Comment:
Maybe the behavior should be different here than the Lexer.
In the lexer its easy to calculate endOffset if we use "onlyOffset" because we just add the token's length.

But in the CSTNode it won't be easy to calculate the endOffset from the startOffset.

Reply:
We should probably give different names than the lexer modes to avoid confusion

Reply:
So you want 3 options, which can be combined?

- One with start/end info on offset
- One with start/end info on line
- One with start/end info on column

Also, I'm not sure on how to implement it : for instance, if we only have the startOffset info, should we compute the endOffset by getting the last token startOffset and adding its length ? It would be more difficult to compute the endColumn and endLine, if we don't have this info in the tokens

Reply:
Lets simplify things so we can finish this feature 😄 

1. Lets have only two options
   - Offsets (start + end)
   - Full info

2. Lets document an assumption that if you want Full info you need to enable full location tracking in the lexer too.

Reply:
LGTM :smile: